custom-wiki-links.ts

```diff
@@ -0,0 +1,98 @@
+// Custom remark plugin to handle wiki links with proper path resolution
+
+export function customWikiLinks() {
+  const searchDirs = ["zk", "zk/Register"];
+  const baseDir = ".";
+  
+  function resolveLinkPath(link: string): string {
+    // Use Deno's file system API
+    try {
+      for (const searchDir of searchDirs) {
+        const fullPath = `${baseDir}/${searchDir}/${link}.md`;
+        try {
+          Deno.statSync(fullPath);
+          return `/${searchDir}/${link}/`;
+        } catch {
+          // File doesn't exist, continue to next directory
+        }
+      }
+    } catch {
+      // Fallback if Deno API fails
+    }
+    
+    // If not found, return the original link (fallback)
+    return `/${link}/`;
+  }
+
+  function processNode(node: any): void {
+    if (node.type === 'text' && typeof node.value === 'string') {
+      const wikiLinkRegex = /\[\[([^\]]+)\]\]/g;
```

Comment:
The remark-obsidian also supports the following syntax: `[[Internal link|With custom text]]`, `[[Internal link#heading]]`, `[[Internal link#heading|With custom text]]`.